src/assets/scripts/client/aircraft/FlightManagementSystem/WaypointModel.js

```diff
@@ -368,6 +368,32 @@ export default class WaypointModel {
     }
 
     /**
+     * Check for a maximum altitude restriction below the given altitude
+     *
+     * @for WaypointModel
+     * @method hasMaximumAltitudeBelow
+     * @param altitude {number} in feet
+     * @return {boolean}
+     */
+    hasMaximumAltitudeBelow(altitude) {
```

Comment:
At first glance, I feel like these should be just `.hasMaximumAltitude()`. It would seem strange to have `.hasMaximumAltitudeBelow(altitude)` without a way to see if there simply is one or not. However, we do get the benefit here of excluding any restrictions that are above the aircraft's cruise altitude, which is something I wouldn't have thought of and would definitely want to keep in some form. Maybe this _is_ best to keep it this way? :thinking:

Reply:
I suspect there is a more elegant solution along the lines of a `.hasMaximumAltitude()` and `.hasMaximumAltitudeFiltered()` but I can't come up with the right words to convey the semantics of such a thing.  For now it's probably a good compromise.

Reply:
Works for now! :smiley: